### ISSUE TITLE
Flash of unstyled content when selecting a menu item

### ISSUE THREAD

**Issue by yatil** [w3c/web-roadmaps#228]
When selecting a menu item, the page sometimes only flashes unstyled content before loading all the surrounding data and CSS. That’s weird. (Observed in latest Safari on latest MacOS.

See video here: https://cl.ly/3a2O1h40133r

**Comment by tidoust** [comment]
Thanks for the report (and video!), @yatil. That's a known and unfortunate consequence of the fact that the HTML page is assembled client-side for now: the browser first loads the raw content and then the right HTML template and CSS styles, which can take up to a few seconds and trigger the nasty flash.

That problem does not exist in published snapshots, which contain generated pages.

While the generation of a given page is automatic, the generation of an entire snapshot is still a manual process for now. What I'd like to do is to automate that generation so that the pages that get served on GitHub Pages can become the generated versions.

That supposes also fixing the content negotiation issue noted in https://github.com/w3c/web-roadmaps/issues/68#issuecomment-338312178 for translations though...

Anyway, we know the issue exists, and fixing it properly will take some time. I'll see if I can find a quick workaround in the meantime.

**Comment by tidoust** [comment]
> That supposes also fixing the content negotiation issue noted in #68 (comment) for translations though...

Actually, there should be no need to fix that issue for now. If we stick to current naming convention, we just won't do any content negotiation on the version published through GitHub Pages. People will just always get the English version, with a possibility to switch to existing translations from the footer.

**Comment by yatil** [comment]
Ah, OK, thanks for the clarification!

**Comment by tidoust** [comment]
My last comment may be a bit unclear. What I meant is that there is no need to fix the content negotiation issue, but the flashing effect should still be fixed, so keeping the issue open seems a good thing :)

**Comment by tidoust** [comment]
Pages published on GitHub Pages now are the fully generated versions of the pages which:
1. load faster (they don't need to fetch HTML templates and spec data)
2. do not "flash"

Technically, the flashing effect still exists in development mode, meaning when a roadmap author edits the pages locally on her computer before committing a change. That's not fantastic, but I think we can live with it for now. Closing the issue accordingly.
